aws/src/main/java/org/apache/iceberg/aws/s3/S3InputFile.java

```diff
@@ -82,6 +97,27 @@ public long getLength() {
 
   @Override
   public SeekableInputStream newStream() {
+    if (s3FileIOProperties().isS3AnalyticsAcceleratorEnabled()) {
+      LOG.info("Using S3 analytics accelerator stream");
+      LOG.info("S3 analytics accelerator configuration: {}", streamFactory().getConfiguration());
+      try {
+        final ObjectMetadata metadata =
+            ObjectMetadata.builder()
+                .contentLength(getObjectMetadata().contentLength())
+                .etag(getObjectMetadata().eTag())
+                .build();
+        return new AnalyticsAcceleratorInputStream(
+            streamFactory()
+                .createStream(
+                    software.amazon.s3.analyticsaccelerator.util.S3URI.of(
+                        uri().bucket(), uri().key()),
+                    metadata));
+      } catch (IOException e) {
+        throw new RuntimeIOException(
+            "Failed to create analytics accelerator input stream for bucket: %s, key: %s - %s",
+            uri().bucket(), uri().key(), e);
+      }
+    }
     return new S3InputStream(client(), uri(), s3FileIOProperties(), metrics());
```

Comment:
Since we already have an input stream factory, I think we could avoid having a branching logic when creating a new stream. Instead, the default input stream factory would just be using the `S3Client` to create a new input stream. Only when the flag is enabled, you use a different input stream factory.